index.d.ts

```diff
@@ -1,66 +1,257 @@
+	 */
+	implicitRender: () => void
+
+	/**
+	 * Excecutes the challenge after the render() function has been called, by invoking the
+	 * turnstile.execute(container: string | HTMLElement, jsParams?: RenderParameters) function
+	 * separately, decoupling the appearance and rendering of a widget from its execution.
+	 * @see [Execution Modes](https://developers.cloudflare.com/turnstile/get-started/client-side-rendering/#execution-modes)
+	 */
+	execute: () => void
+
+	/**
+	 * Explicitly renders the Turnstile widget.  The `sitekey` parameter is required. The
+	 * `callback` parameter, however, is only required if `execution` is set to `execute`.
+	 * @param container - HTML container to render the widget.
+	 * @param options - Configuration options for rendering. See {@link RenderOptions}.
+	 * @returns A `widgetId` string if successful, otherwise undefined.
+	 */
+	render(container: HTMLElement | string, options: RenderOptions): string | undefined
+
+	/**
+	 * Resets the widget.
+	 * @param widgetId - The ID of the widget.
+	 */
+	reset(widgetId: WidgetId): void
+
+	/**
+	 * Removes the widget from the page.
+	 * @param widgetId - The ID of the widget.
+	 */
+	remove(widgetId: WidgetId): void
+
+	/**
+	 * Obtains the widget's response using its widgetId.
+	 * @param widgetId - The ID of the widget.
+	 */
+	getResponse(widgetId: WidgetId): string
+
+	/**
+	 * Checks if the widget has expired.
+	 * @param widgetId - The ID of the widget.
+	 */
+	isExpired(widgetId: WidgetId): boolean
 }
 
-export interface TurnstileOptions {
-	sitekey: string;
-	action?: string;
-	cData?: string;
-	callback?: (token: string) => void;
-	"error-callback"?: () => void;
-	"expired-callback"?: (token: string) => void;
-	"timeout-callback"?: () => void;
-	"before-interactive-callback"?: () => void;
-	"after-interactive-callback"?: () => void;
-	"unsupported-callback"?: () => void;
-	theme?: "light" | "dark" | "auto"; // defaults to "auto"
-	language?: SupportedLanguages | "auto"; // defaults to "auto"
-	tabindex?: number; // defaults to 0
-	"response-field"?: boolean; // defaults to true
-	"response-field-name"?: string; // defaults to "cf-turnstile-response"
-	size?: "normal" | "invisible" | "compact"; // defaults to "normal"
-	retry?: "auto" | "never"; // defaults to "auto"
-	"retry-interval"?: number; // up to 15m (900_000) in ms, defaults to 8s
-	"refresh-expired"?: "auto" | "manual" | "never"; // defaults to "auto"
-	appearance?: "always" | "execute" | "interaction-only";
-	execution?: "render" | "execute";
-
-	chlPageData?: string; // ENTERPRISE ONLY - undocumented
+export type TurnstileOptions = RenderOptions
+
+/**
+ * Interface for Turnstile rendering parameters.
+ */
+export interface RenderOptions {
+	/**
+	 * Every widget has a sitekey. This sitekey is associated with the corresponding
+	 * widget configuration and is created upon the widget creation.
```

Comment:
> and is created upon the widget creation

It's not obvious that this is referring the to the creation on the dashboard (especially because Widget is a client-side term, the dashboard is referring to it as "Sites")